lib/check_params.h

```diff
@@ -918,6 +918,62 @@ void printQudaGaugeObservableParam(QudaGaugeObservableParam *param)
 #endif
 }
 
+#if defined INIT_PARAM
+QudaBLASParam newQudaBLASParam(void)
+{
+  QudaBLASParam ret;
+#elif defined CHECK_PARAM
+static void checkBLASParam(QudaBLASParam *param)
+{
+#else
+void printQudaBLASParam(QudaBLASParam *param)
+{
+  printfQuda("QUDA blas parameters:\n");
+#endif
+
+#ifdef INIT_PARAM
+  P(trans_a, QUDA_BLAS_OP_N);
+  P(trans_b, QUDA_BLAS_OP_N);
+  P(m, 0);
+  P(n, 0);
+  P(k, 0);
+  P(lda, 0);
+  P(ldb, 0);
+  P(ldc, 0);
+  P(a_offset, 0);
+  P(b_offset, 0);
+  P(c_offset, 0);
+  P(strideA, -1);
+  P(strideB, -1);
+  P(strideC, -1);
+  P(batch_count, 1);
+  P(data_type, QUDA_BLAS_DATATYPE_S);
+  P(data_order, QUDA_BLAS_DATAORDER_ROW);
+#else
+  P(trans_a, QUDA_BLAS_OP_INVALID);
+  P(trans_b, QUDA_BLAS_OP_INVALID);
+  P(m, INVALID_INT);
+  P(n, INVALID_INT);
+  P(k, INVALID_INT);
+  P(lda, INVALID_INT);
+  P(ldb, INVALID_INT);
+  P(ldc, INVALID_INT);
+  P(a_offset, INVALID_INT);
+  P(b_offset, INVALID_INT);
+  P(c_offset, INVALID_INT);
+  P(strideA, INVALID_INT);
+  P(strideB, INVALID_INT);
+  P(strideC, INVALID_INT);
+  P(batch_count, INVALID_INT);
+  P(data_type, QUDA_BLAS_DATATYPE_INVALID);
+  P(data_order, QUDA_BLAS_DATAORDER_INVALID);
+#endif
```

Comment:
Is the `INIT_PARAM` specialization needed here?  If all of the above parameters need to be set, and it looks like most of the initialized values aren't  valid for computation, e.g., `m = n = k = 0`, then just going with the `INVALID_INT` initialization is likely better here I think.